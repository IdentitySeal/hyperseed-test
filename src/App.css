 .app__container{
  margin: 0;
  background-color: #2196F3;
  padding: 10px;
 }
