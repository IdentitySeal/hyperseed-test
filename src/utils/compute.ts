
interface IcomputeArgs {
    x:number
    y:number
}

export class computeService {

      static addXandY = (props:IcomputeArgs) => {
        return (props.x + props.y)
      }
      static multiplyXandY = (props:IcomputeArgs)=> {
        return (props.x * props.y)

      }
      static powerXtoY = (props:IcomputeArgs)=> {
        return (props.x ^ props.y)
      }
}
