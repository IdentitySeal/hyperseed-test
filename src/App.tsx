import React from 'react';
import './App.css';
import RowAdder from './component/RowAdder/RowAdder';


function App() {
  return (
    <div className='app__container'>
    <RowAdder/>
    </div>
  );
}

export default App;
