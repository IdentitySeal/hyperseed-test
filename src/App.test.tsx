import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import { computeService } from './utils/compute';


it('renders correctly ', () => {
  const wrapper =   render(<App />);

  expect((wrapper)).toMatchSnapshot();
});

describe('Testing Utils Opertions', () => {
  it('should equal 4',()=>{
     expect(computeService.addXandY({x:2,y:2})).toBe(4);
    })

  test('It should equal 12', () => {
    expect(computeService.multiplyXandY({x:2,y:6})).toBe(12);
  }) 

  test('It should equal 1', () => {
    expect(computeService.powerXtoY({x:4,y:5})).toBe(1);
  }) 
});
