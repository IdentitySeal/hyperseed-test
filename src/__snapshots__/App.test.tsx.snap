// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app__container"
      >
        <div
          class="App"
        >
          <div
            class=""
          >
            <div
              class="container header"
            >
              <div>
                X
              </div>
              <div>
                Y
              </div>
              <div>
                X + Y
              </div>
              <div>
                X * Y
              </div>
              <div>
                X ^ Y
              </div>
            </div>
            <div
              class=""
            >
              <div
                class="container"
              >
                <div
                  class="editable__box"
                  contenteditable="true"
                />
                <div
                  class="editable__box"
                  contenteditable="true"
                />
                <div
                  class="editable__box"
                  contenteditable="false"
                />
                <div
                  class="editable__box"
                  contenteditable="false"
                />
                <div
                  class="editable__box"
                  contenteditable="false"
                />
              </div>
              <div
                class="button__container"
              />
              <button
                class=""
              >
                Add Row
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app__container"
    >
      <div
        class="App"
      >
        <div
          class=""
        >
          <div
            class="container header"
          >
            <div>
              X
            </div>
            <div>
              Y
            </div>
            <div>
              X + Y
            </div>
            <div>
              X * Y
            </div>
            <div>
              X ^ Y
            </div>
          </div>
          <div
            class=""
          >
            <div
              class="container"
            >
              <div
                class="editable__box"
                contenteditable="true"
              />
              <div
                class="editable__box"
                contenteditable="true"
              />
              <div
                class="editable__box"
                contenteditable="false"
              />
              <div
                class="editable__box"
                contenteditable="false"
              />
              <div
                class="editable__box"
                contenteditable="false"
              />
            </div>
            <div
              class="button__container"
            />
            <button
              class=""
            >
              Add Row
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
